src/components/TransactionHistoryRows/TransactionContract.tsx

```diff
@@ -21,6 +23,12 @@ export const TransactionContract = ({
   let assets = []
   if (txDetails.sellAsset) assets.push(parseRelevantAssetFromTx(txDetails, AssetTypes.Source))
   if (txDetails.buyAsset) assets.push(parseRelevantAssetFromTx(txDetails, AssetTypes.Destination))
+  const isReceive = txDetails.tradeTx?.type === TxType.Receive
+  const interactsWithWithdrawMethod = txDetails.tx.data?.method === ContractMethod.Withdraw
+  const isSend = txDetails.tradeTx?.type === TxType.Send
+  const i18n = isReceive ? txDetails.tradeTx?.type : txDetails.tx.data?.method
+  const isFirstAssetOutgoing = interactsWithWithdrawMethod && isSend
```

Comment:
My naming suggestion may have led you astray here - it should probably be `isAssetOutgoing`, as it's not always the first asset, yeah?

Indeed, you check if it is first in the generic row (`prefix={index === 0 && isFirstAssetOutgoing ? '-' : ''}`).

Reply:
It will always be the first asset in the components assets list (and whats displayed) but not guaranteed to be the first asset in the transfers list where they come from.